addons/im_livechat/models/chatbot_mail_message.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class ChatbotMailMessage(models.Model):
+    """ Chatbot Mail Message
+        We create a new model to store the related step to a mail.message and the user's answer.
+        We do this in a new model to avoid bloating the 'mail.message' model.
+    """
+
+    _name = 'im_livechat.chatbot.mail.message'
+    _description = 'Chatbot Mail Message'
+
+    mail_message_id = fields.Many2one('mail.message', string='Related Mail Message')
```

Comment:
As discussed, could be great to enforce the link (make it required). The "welcome message" trick would be to create the mail.message related to the welcome message when the customer answers. That way we have a message and we can have a coherent db content, always having a message linked to a chatbot message. This allows to remove ``if-else`` constructs linked to that specific first step.

Reply:
The "welcome steps" are not automatically posted into the channel on the first user interaction with the bot.
This required a bit of added logic but seems to work well and allows to simplify the model as you suggested.

See "_post_welcome_steps".